client/state/signup/progress/actions.js

```diff
@@ -64,6 +64,10 @@ function recordSubmitStep( stepName, providedDependencies ) {
 					.join( ',' );
 			}
 
+			if ( includes( [ 'selected_design' ], propName ) ) {
+				propValue = propValue.slug;
```

Comment:
@razvanpapadopol is there a reason why, in your draft PR, you didn't pass the `selectedDesign.slug` in [`this.props.submitSignupStep`](https://github.com/Automattic/wp-calypso/pull/51501/files#diff-425a2aeab314234f95622905026eef2b94eeafa1ecbd493c5bf3d519d44757afR42-R50), instead of transforming it after?

Maybe I'm missing something

Reply:
I was thinking that we need to pass `selectedDesign` object in `providedDependencies` for its values to be available in `client/lib/signup/step-actions/index.js` but if there is a way to pass just the slug for recording purposes, that's fine.

Reply:
You're probably right. `selectedDesign` as a dependency needs to provided in full (ie. the full object) for the other step actions to work (e.g. adding new site parameters)